### PR TITLE
fix(coq-serapi): cmdliner<1.1.0

### PR DESCRIPTION
**Pull request by erikmd**
**TL;DR:** some releases of coq-serapi [FTBFS](https://gitlab.com/coq-community/docker-coq/-/jobs/2284076471) with the latest cmdliner releases; hence this PR that merely adds upper-bounds.

For details, see: https://github.com/ejgallego/coq-serapi/issues/267

Cc @ejgallego, do you approve this PR?

### CODE REVIEW

**Comment by erikmd** [comment]
Hi @kit-ty-kate, thanks!

I'm not familiar with the interface of http://check.ocamllabs.io/
but according to your feedback, I reverted the `cmdliner` upper-bound for `≥ 8.9*`

Also, I took into account the feedback from [opam-ci](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/b90d9199f0c021f1fdb69c5b88df18b0e6fa690c), especially regarding missing lower-bounds (or an upper one for `ppx_deriving`).

Let's see what the CI will say… then let me know if I have to tweak this PR further :)

**Comment by erikmd** [comment]
FTR, we now get 8 similar failures:

* 4.06
        [coq-serapi.8.7.1+0.4 (failed: File "ppx_deriving_map.cppo.ml", line 1, characters 0-33)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/602872b78009e642a8506b73c003db39e7a2f8e9/variant/opam-2.1,compilers,4.06,coq-serapi.8.7.1+0.4)
        [coq-serapi.8.7.1+0.4.12 (failed: File "ppx_deriving_map.cppo.ml", line 1, characters 0-33)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/602872b78009e642a8506b73c003db39e7a2f8e9/variant/opam-2.1,compilers,4.06,coq-serapi.8.7.1+0.4.12)
        [coq-serapi.8.8.0+0.5.4 (failed: File "ppx_deriving_map.cppo.ml", line 1, characters 0-33)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/602872b78009e642a8506b73c003db39e7a2f8e9/variant/opam-2.1,compilers,4.06,coq-serapi.8.8.0+0.5.4)
*    4.07
        [coq-serapi.8.8.0+0.5.4 (failed: File "ppx_deriving_map.cppo.ml", line 1, characters 0-33)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/602872b78009e642a8506b73c003db39e7a2f8e9/variant/opam-2.1,compilers,4.07,coq-serapi.8.8.0+0.5.4)
*    4.08
        coq-serapi.8.8.0+0.5.2 → [lower-bounds (failed: File "ppx_deriving_map.cppo.ml", line 1, characters 0-33)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/602872b78009e642a8506b73c003db39e7a2f8e9/variant/opam-2.1,compilers,4.08,coq-serapi.8.8.0+0.5.2,lower-bounds)
        [coq-serapi.8.8.0+0.5.4 (failed: File "ppx_deriving_map.cppo.ml", line 1, characters 0-33)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/602872b78009e642a8506b73c003db39e7a2f8e9/variant/opam-2.1,compilers,4.08,coq-serapi.8.8.0+0.5.4)
*    4.09
        [coq-serapi.8.8.0+0.5.4 (failed: File "ppx_deriving_map.cppo.ml", line 1, characters 0-33)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/602872b78009e642a8506b73c003db39e7a2f8e9/variant/opam-2.1,compilers,4.09,coq-serapi.8.8.0+0.5.4)
        [coq-serapi.8.8.0+0.5.5 (failed: File "ppx_deriving_map.cppo.ml", line 1, characters 0-33)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/602872b78009e642a8506b73c003db39e7a2f8e9/variant/opam-2.1,compilers,4.09,coq-serapi.8.8.0+0.5.5)

Ideas?

(it seems that imposing `"ppx_deriving" < "5.0"` [what I did for coq-serapi.8.8.0+0.5.5](https://github.com/ocaml/opam-repository/pull/21099/commits/b6b27dfecec9d4472ba5aabe5693c8112d1ccc6b) is not enough)

**Comment by erikmd** [comment]
On second thought, the remaining issue sounded like the one fixed by https://github.com/ocaml/opam-repository/pull/20145
Only the versions {4.4, 4.4.1} of `ppx_deriving` should be impacted by the `dune` 3.0 compat. issue
So, next round of CI (I also pushed a tiny commit to fix the linting error)

**Comment by erikmd** [comment]
Now, it sounds better… − IIUC, only lower-bounds issues regarding `ppx_deriving`.{4.4, 4.4.1} (which wasn't the intended package to update in the first place :)

@kit-ty-kate WDYT?

**Comment by erikmd** [comment]
Hi @kit-ty-kate − just a friendly ping 🙂
Do you believe the PR is acceptable as is?

**Comment by kit-ty-kate** [comment]
Pasting the errors linked to the new lower-bounds for safe-keeping:
```
#=== ERROR while compiling coq-serapi.8.7.1+0.4 ===============================#
# context              2.1.2 | linux/x86_64 | ocaml-base-compiler.4.06.1 | pinned(https://github.com/ejgallego/coq-serapi/archive/8.7.1+0.4.tar.gz)
# path                 ~/.opam/4.06/.opam-switch/build/coq-serapi.8.7.1+0.4
# command              ~/.opam/opam-init/hooks/sandbox.sh build make -j31 TARGET=native
# exit-code            2
# env-file             ~/.opam/log/coq-serapi-36814-182e55.env
# output-file          ~/.opam/log/coq-serapi-36814-182e55.out
### output ###
# OCAMLFIND_IGNORE_DUPS_IN=/home/opam/.opam/4.06/lib/ocaml/compiler-libs/ \
# OCAMLPATH=                              \
# ocamlbuild -use-ocamlfind -j 4   sertop/sertop.native
# ocamlfind ocamldep -package cmdliner -package str -package coq.intf -package coq.ltac -package sexplib -package coq.stm -package unix -package threads -package num -modules sertop/sertop.ml > sertop/sertop.ml.depends
# ocamlfind ocamldep -package str -package coq.intf -package coq.ltac -package sexplib -package coq.stm -package unix -package threads -package num -modules sertop/sertop_init.mli > sertop/sertop_init.mli.depends
# ocamlfind ocamldep -package str -package coq.intf -package coq.ltac -package sexplib -package coq.stm -package unix -package threads -package num -modules sertop/sertop_sexp.mli > sertop/sertop_sexp.mli.depends
# ocamlfind ocamldep -package str -package coq.intf -package coq.ltac -package sexplib -package unix -package threads -package num -modules serapi/serapi_protocol.mli > serapi/serapi_protocol.mli.depends
# ocamlfind ocamldep -package str -package coq.intf -package coq.ltac -package unix -package threads -package num -modules serapi/serapi_goals.mli > serapi/serapi_goals.mli.depends
# ocamlfind ocamlc -c -g -annot -bin-annot -safe-string -rectypes -thread -package str -package coq.intf -package coq.ltac -w -58 -w +1..49-4-44 -warn-error +1..49-42 -package unix -package threads -package num -I serapi -I sertex -I serlib -I coq-libjs -I serlib/ide -I serlib/plugins/ltac -o serapi/serapi_goals.cmi serapi/serapi_goals.mli
# ocamlfind ocamlc -c -g -annot -bin-annot -safe-string -rectypes -thread -package str -package coq.intf -package coq.ltac -package sexplib -package coq.stm -w -58 -w +1..49-4-44 -warn-error +1..49-42 -package unix -package threads -package num -I sertop -I sertex -I serlib -I serapi -I coq-libjs -I serlib/ide -I serlib/plugins/ltac -o sertop/sertop_init.cmi sertop/sertop_init.mli
# ocamlfind ocamlc -c -g -annot -bin-annot -safe-string -rectypes -thread -package str -package coq.intf -package coq.ltac -package sexplib -w -58 -w +1..49-4-44 -warn-error +1..49-42 -package unix -package threads -package num -I serapi -I sertex -I serlib -I coq-libjs -I serlib/ide -I serlib/plugins/ltac -o serapi/serapi_protocol.cmi serapi/serapi_protocol.mli
# ocamlfind ocamlc -c -g -annot -bin-annot -safe-string -rectypes -thread -package str -package coq.intf -package coq.ltac -package sexplib -package coq.stm -w -58 -w +1..49-4-44 -warn-error +1..49-42 -package unix -package threads -package num -I sertop -I sertex -I serlib -I serapi -I coq-libjs -I serlib/ide -I serlib/plugins/ltac -o sertop/sertop_sexp.cmi sertop/sertop_sexp.mli
# ocamlfind ocamlc -c -g -annot -bin-annot -safe-string -rectypes -thread -package cmdliner -package str -package coq.intf -package coq.ltac -package sexplib -package coq.stm -w -58 -w +1..49-4-44 -warn-error +1..49-42 -package unix -package threads -package num -I sertop -I sertex -I serlib -I serapi -I coq-libjs -I serlib/ide -I serlib/plugins/ltac -o sertop/sertop.cmo sertop/sertop.ml
# + ocamlfind ocamlc -c -g -annot -bin-annot -safe-string -rectypes -thread -package cmdliner -package str -package coq.intf -package coq.ltac -package sexplib -package coq.stm -w -58 -w +1..49-4-44 -warn-error +1..49-42 -package unix -package threads -package num -I sertop -I sertex -I serlib -I serapi -I coq-libjs -I serlib/ide -I serlib/plugins/ltac -o sertop/sertop.cmo sertop/sertop.ml
# File "sertop/sertop.ml", line 44, characters 21-33:
# Error: Unbound value Arg.doc_alts
# Command exited with code 2.
# make: *** [Makefile:29: sertop] Error 10
```
```
#=== ERROR while compiling coq-serapi.8.7.1+0.4.12 ============================#
# context              2.1.2 | linux/x86_64 | ocaml-base-compiler.4.06.1 | pinned(https://github.com/ejgallego/coq-serapi/archive/8.7.2+0.4.12.tar.gz)
# path                 ~/.opam/4.06/.opam-switch/build/coq-serapi.8.7.1+0.4.12
# command              ~/.opam/opam-init/hooks/sandbox.sh build make -j31 TARGET=native
# exit-code            2
# env-file             ~/.opam/log/coq-serapi-36860-b4fe1f.env
# output-file          ~/.opam/log/coq-serapi-36860-b4fe1f.out
### output ###
# fatal: not a git repository (or any parent up to mount point /)
# Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
# echo "let ser_git_version = \"0.4.10\";;" > sertop/ser_version.ml
# OCAMLFIND_IGNORE_DUPS_IN=/home/opam/.opam/4.06/lib/ocaml/compiler-libs/ \
# OCAMLPATH=                              \
# ocamlbuild -use-ocamlfind -j 4   sertop/sertop.native
# ocamlfind ocamldep -package cmdliner -package coq.intf,coq.stm,coq.ltac,sexplib -modules sertop/sertop.ml > sertop/sertop.ml.depends
# ocamlfind ocamldep -package coq.intf,coq.stm,coq.ltac,sexplib -modules sertop/ser_version.ml > sertop/ser_version.ml.depends
# ocamlfind ocamldep -package sexplib,coq.intf,coq.stm -modules serlib/serlib_init.mli > serlib/serlib_init.mli.depends
# ocamlfind ocamldep -package cmdliner -package coq.intf,coq.stm,coq.ltac,sexplib -modules sertop/sertop_arg.mli > sertop/sertop_arg.mli.depends
# ocamlfind ocamldep -package coq.intf,coq.stm,coq.ltac,sexplib -modules sertop/sertop_init.mli > sertop/sertop_init.mli.depends
# ocamlfind ocamldep -package coq.intf,coq.stm,coq.ltac,sexplib -modules sertop/sertop_ser.mli > sertop/sertop_ser.mli.depends
# ocamlfind ocamldep -package sexplib -package coq.intf,coq.ltac -modules serapi/serapi_protocol.mli > serapi/serapi_protocol.mli.depends
# ocamlfind ocamldep -package coq.intf,coq.ltac -modules serapi/serapi_goals.mli > serapi/serapi_goals.mli.depends
# ocamlfind ocamlc -c -g -annot -bin-annot -safe-string -rectypes -thread -package coq.intf,coq.ltac -w +1..49-4-44-45 -warn-error +1..49-42 -I serapi -I sertex -I serlib -I serlib/ide -I serlib/plugins/ltac -I jscoq/coq-js -o serapi/serapi_goals.cmi serapi/serapi_goals.mli
# ocamlfind ocamlc -c -g -annot -bin-annot -safe-string -rectypes -thread -package sexplib -package coq.intf,coq.ltac -w +1..49-4-44-45 -warn-error +1..49-42 -I serapi -I sertex -I serlib -I serlib/ide -I serlib/plugins/ltac -I jscoq/coq-js -o serapi/serapi_protocol.cmi serapi/serapi_protocol.mli
# ocamlfind ocamlc -c -g -annot -bin-annot -safe-string -rectypes -thread -package coq.intf,coq.stm,coq.ltac,sexplib -w +1..49-4-44-45 -warn-error +1..49-42 -I sertop -I sertex -I serlib -I serapi -I serlib/ide -I serlib/plugins/ltac -I jscoq/coq-js -o sertop/sertop_init.cmi sertop/sertop_init.mli
# ocamlfind ocamlc -c -g -annot -bin-annot -safe-string -rectypes -thread -package coq.intf,coq.stm,coq.ltac,sexplib -w +1..49-4-44-45 -warn-error +1..49-42 -I sertop -I sertex -I serlib -I serapi -I serlib/ide -I serlib/plugins/ltac -I jscoq/coq-js -o sertop/sertop_ser.cmi sertop/sertop_ser.mli
# ocamlfind ocamldep -package coq.intf,coq.stm,coq.ltac,sexplib -modules sertop/sertop_sexp.mli > sertop/sertop_sexp.mli.depends
# ocamlfind ocamlc -c -g -annot -bin-annot -safe-string -rectypes -thread -package coq.intf,coq.stm,coq.ltac,sexplib -w +1..49-4-44-45 -warn-error +1..49-42 -I sertop -I sertex -I serlib -I serapi -I serlib/ide -I serlib/plugins/ltac -I jscoq/coq-js -o sertop/ser_version.cmo sertop/ser_version.ml
# ocamlfind ocamlc -c -g -annot -bin-annot -safe-string -rectypes -thread -package sexplib,coq.intf,coq.stm -w +1..49-4-44-45 -warn-error +1..49-42 -I serlib -I sertex -I serapi -I serlib/ide -I serlib/plugins/ltac -I jscoq/coq-js -o serlib/serlib_init.cmi serlib/serlib_init.mli
# ocamlfind ocamlc -c -g -annot -bin-annot -safe-string -rectypes -thread -package cmdliner -package coq.intf,coq.stm,coq.ltac,sexplib -w +1..49-4-44-45 -warn-error +1..49-42 -I sertop -I sertex -I serlib -I serapi -I serlib/ide -I serlib/plugins/ltac -I jscoq/coq-js -o sertop/sertop_arg.cmi sertop/sertop_arg.mli
# ocamlfind ocamlc -c -g -annot -bin-annot -safe-string -rectypes -thread -package coq.intf,coq.stm,coq.ltac,sexplib -w +1..49-4-44-45 -warn-error +1..49-42 -I sertop -I sertex -I serlib -I serapi -I serlib/ide -I serlib/plugins/ltac -I jscoq/coq-js -o sertop/sertop_sexp.cmi sertop/sertop_sexp.mli
# ocamlfind ocamlc -c -g -annot -bin-annot -safe-string -rectypes -thread -package cmdliner -package coq.intf,coq.stm,coq.ltac,sexplib -w +1..49-4-44-45 -warn-error +1..49-42 -I sertop -I sertex -I serlib -I serapi -I serlib/ide -I serlib/plugins/ltac -I jscoq/coq-js -o sertop/sertop.cmo sertop/sertop.ml
# + ocamlfind ocamlc -c -g -annot -bin-annot -safe-string -rectypes -thread -package cmdliner -package coq.intf,coq.stm,coq.ltac,sexplib -w +1..49-4-44-45 -warn-error +1..49-42 -I sertop -I sertex -I serlib -I serapi -I serlib/ide -I serlib/plugins/ltac -I jscoq/coq-js -o sertop/sertop.cmo sertop/sertop.ml
# File "sertop/sertop.ml", line 57, characters 8-13:
# Error: Unbound value const
# Hint: Did you mean cos or cosh?
# Command exited with code 2.
# make: *** [Makefile:34: sertop] Error 10
```

File: packages/coq-serapi/coq-serapi.8.7.1+0.4.1/opam [hunk 1]
```diff
@@ -14,7 +14,7 @@ depends: [
   "ocamlbuild" {build}
   "ppx_import" {>= "1.4"}
   "ppx_deriving" {>= "4.2.1"}
-  "cmdliner"
+  "cmdliner" {>= "0.9.8" & < "1.1.0" }
```
Comment (kit-ty-kate) [inline packages/coq-serapi/coq-serapi.8.7.1+0.4.1/opam:17]:
```suggestion
  "cmdliner" {>= "0.9.6" & < "1.1.0" }
```

File: packages/coq-serapi/coq-serapi.8.7.1+0.4.2/opam [hunk 2]
```diff
@@ -10,7 +10,7 @@ depends: [
   "ocaml" {>= "4.06.0"}
   "coq" {>= "8.7.1+1" & < "8.8"}
   "camlp5"
-  "cmdliner"
+  "cmdliner" {>= "0.9.8" & < "1.1.0"}
```
Comment (kit-ty-kate) [inline packages/coq-serapi/coq-serapi.8.7.1+0.4.2/opam:13]:
```suggestion
  "cmdliner" {>= "0.9.6" & < "1.1.0"}
```

File: packages/coq-serapi/coq-serapi.8.7.1+0.4/opam [hunk 3]
```diff
@@ -14,7 +14,7 @@ depends: [
   "ocamlbuild" {build}
   "ppx_import" {>= "1.4"}
   "ppx_deriving" {>= "4.2.1"}
-  "cmdliner"
+  "cmdliner" {>= "0.9.8" & < "1.1.0"}
```
Comment (kit-ty-kate) [inline packages/coq-serapi/coq-serapi.8.7.1+0.4/opam:17]:
```suggestion
  "cmdliner" {>= "0.9.6" & < "1.1.0"}
```

**Review by kit-ty-kate — COMMENTED** [review]
`Arg.doc_alts` was added in 0.9.6 not 0.9.8

**Comment by erikmd** [comment]
> `Arg.doc_alts` was added in 0.9.6 not 0.9.8

Indeed, but `Arg.const` was added in 0.9.8:

https://github.com/dbuenzli/cmdliner/commit/f81ab2a858c4ab6d6b75ab3e54295d4bc44d134a

So, not sure that we can downgrade to 0.9.6… we'll see with the CI anyway ^^

**Comment by ejgallego** [comment]
Thanks a lot folks for all the help with this.

**Comment by kit-ty-kate** [comment]
Thanks!

**Pull request merged**